pennylane/templates/subroutines/approx_time_evolution.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``TimeEvolution`` template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+import pennylane as qml
+from pennylane.templates.decorator import template
+from pennylane.wires import Wires
+
+
+@template
+def ApproxTimeEvolution(hamiltonian, wires, time, n):
+    r""" Applies the Trotterized time-evolution operator for an arbitrary Hamiltonian, expressed in terms
+    of Pauli gates. The general
+    time-evolution operator for a time-independent Hamiltonian is given by:
+
+    .. math:: U(t) \ = \ e^{-i H t},
+
+    for some Hamiltonian of the form:
+
+    .. math:: H \ = \ \displaystyle\sum_{n} H_n.
```

Comment:
```suggestion
    .. math:: H \ = \ \displaystyle\sum_{j} H_j.
```
To not confuse with `n` as in the number of Trotter steps